v2/internal/frontend/desktop/linux/window.go

```diff
@@ -635,6 +648,7 @@ func NewWindow(appoptions *options.App, debug bool) *Window {
 	result.SetTitle(appoptions.Title)
 	result.SetMinSize(appoptions.MinWidth, appoptions.MinHeight)
 	result.SetMaxSize(appoptions.MaxWidth, appoptions.MaxHeight)
+	result.SetWindowIcon(appoptions.Linux.Icon)
```

Comment:
```suggestion
	if appoptions.Linux != nil {
		if appoptions.Linux.Icon != nil {
			result.SetWindowIcon(appoptions.Linux.Icon)
		}
	}
```

Reply:
is the second comparison really needed as the length of the slice gets checked in the SetWindowIcon() method?

Reply:
Yeah, this will throw a nil error for apps that don't have a linux specific config but run on linux. It was the first thing that happened when testing this PR :-)

Reply:
sure about the first `if`, I mean the second one: `if appoptions.Linux.Icon != nil {` as this is already tested in `SetWindowIcon`